### ISSUE TITLE
Units have their damage displayed incorrectly in game history

### ISSUE THREAD

**Issue by alkexr** [triplea-game/triplea#4717]
Game history lists units that are present at the start of the battle. If a unit with multiple hitpoints takes damage later in the battle, then the damaged version is shown in the history, retroactively.

Version: at least since 12000

**Comment by VictoryFirst2** [comment]
It looks like the bug is a little more subtle. If you finished a battle with multiple hit units, and you are clicking through the history - if you come from a history line above the battle and then click on the entry that lists the attacker/defender and their units, then the multiple hit units are shown correctly. If you come from below however, from the middle of the battle or from a subsequent phase, that same history line will now show the damaged version of (some of) the multiple hit units. See:

<img width="1919" height="1043" alt="Image" src="https://github.com/user-attachments/assets/aa08b06d-8aea-4bfb-8ebf-07da408c9ff1" />

<img width="1919" height="1043" alt="Image" src="https://github.com/user-attachments/assets/cd6cd862-b22d-4fa4-b8a0-dfce9993ff62" />

You can see that if you come from the other side, it suddenly says 7 damaged battleships and 3 undamaged ones, instead of 10 undamaged battleships.

Save game: [unit damage not displayed correctly.zip](https://github.com/user-attachments/files/31875931/unit.damage.not.displayed.correctly.zip)

**Comment by frigoref** [comment]
@VictoryFirst2 The question is: What is the expected engine behavior?
My understanding would be that if a history entry is selected, you would want to see the state at the end, so after the history entry has happened. Do you agree?
In this case the issue/bug is that  when going forward in the history panel the wrong state (i.e. the history entry's start state) is shown while the end state is expected.
This would resolve the initially described issue of this thread, right?

**Comment by VictoryFirst2** [comment]
@frigoref You are right that when you click on a specific entry, the game should show the map/game state after that action stated in the entry has happened, as with moving units etc.

However, I think for battles it makes less sense. Usually when you click on an entry inside a battle in the history, the map always reflects the state after the battle has happened, which personally I am not a fan of, although I can live with it. But the right-hand side panel showing damaged units when you have selected "Germans attack with X units", does not make sense to me. It is much more logical if that would display all the units that were about to attack in the state they were in before the battle. Germany was not attacking with 7 damaged battleships at the time, but with 10 full-strength battleships.

Therefore, I think the expected behavior should be that order should not matter and the right-hand side panel should always show units in the state they were in at the time, not after the battle (especially when the attackers/defenders are listed).

**Comment by frigoref** [comment]
PR #14956 should fix this issue by making the data change now before displaying/rendering the panel.
Otherwise the data displayed is actually shown based on the source game state as described in this issue.

@VictoryFirst2 I know this is not according to your described expected behavior, but considering that movements also show the result, i.e. the units not being in their source territory, but in their destination territory, I think this is very much consistent this way now.

**Comment by frigoref** [comment]
@alkexr , @VictoryFirst2  Please retest with release [2026-09.16.15773](https://github.com/triplea-game/triplea/releases/tag/2026-09.16.15773) or higher and confirm the issue is resolved.

**Comment by VictoryFirst2** [comment]
@frigoref I tested and the unit images of the save game attached here now always show damaged. So you could say this is a fix, but I still disagree here.

> but considering that movements also show the result, i.e. the units not being in their source territory, but in their destination territory, I think this is very much consistent this way now.

This is true, but there are two different things that reflect the history information. 1) the map 2) the right-hand side panel. It makes sense that 1) the map always shows the state at the end, like when you move something. But that is not applicable for 2) the right-hand side panel. The right-hand side panel telling you this:

<img width="1919" height="676" alt="Image" src="https://github.com/user-attachments/assets/f70c7c6c-c68f-4bf2-87c8-0ca6c3f51637" />

Makes me think that the Germans attacked with 7 damaged battleships and 3 full-strength battleships, while they actually attacked with 10 full-strength battleships. Also, this is not even the state of the units at the end of the battle, this is after the first roll has happened where the British scored 7 hits and damaged 7 battleships. How is this behavior consistent with the rest of the code? 

P.S. To be honest, I also don't like the fact that the map shows the state at the end of the battle all the time when you unfold the battle node and click on any point inside it. Because when you click on the main node, it does not even show the state after the battle (as you said), but rather the start of the battle.

<img width="1105" height="456" alt="Image" src="https://github.com/user-attachments/assets/f595d5fc-7e23-4404-8f2f-338eaeb3b61d" />

And when you click on a node inside the battle node, it suddenly shows the state after the battle has happened

<img width="1124" height="681" alt="Image" src="https://github.com/user-attachments/assets/8621547f-3616-4de5-8a57-3e4421e8b5a1" />

It's better if the map would update along with these nodes as well, so that the territory gradually loses units until the final state is reached when you click on "Germans win" for example. This makes it much easier to follow battles.

But that is a different problem and not about this issue here (which is about the right-hand side panel specifically).
